src/main/java/org/pentaho/di/trans/dataservice/resolvers/TransientResolver.java

```diff
@@ -0,0 +1,150 @@
+  }
+
+  @Override
+  public DataServiceMeta getDataService( String dataServiceName ) {
+    if ( !isTransient( dataServiceName ) ) {
+      return null;
+    }
+    return createDataServiceMeta( dataServiceName );
+  }
+
+  @Override public List<String> getDataServiceNames( String dataServiceName ) {
+    List<String> dataServiceNames = new ArrayList<>();
+    if ( isTransient( dataServiceName ) ) {
+      dataServiceNames.add( dataServiceName );
+    }
+    return dataServiceNames;
+  }
+
+  @Override public List<DataServiceMeta> getDataServices( String dataServiceName, Function<Exception, Void> logger ) {
+    List<DataServiceMeta> dataServiceMetas = new ArrayList<>();
+    if ( isTransient( dataServiceName ) ) {
+      dataServiceMetas.add( createDataServiceMeta( dataServiceName ) );
+    }
+    return dataServiceMetas;
+  }
+
+  @Override public DataServiceExecutor.Builder createBuilder( SQL sql ) {
+    DataServiceMeta dataServiceMeta = getDataService( sql.getServiceName() );
+    if ( dataServiceMeta != null ) {
+      return new DataServiceExecutor.Builder( sql, dataServiceMeta, context );
+    }
+    return null;
+  }
+
+
+  private DataServiceMeta createDataServiceMeta( String dataServiceName ) {
+    String[] parts = dataServiceName.split( DELIMITER );
+    try {
+      String fileAndPath = parts[ 0 ].trim();
+      TransMeta transMeta;
+      Repository repository = repositoryLocator != null ? repositoryLocator.getRepository() : null;
+      if ( repository == null ) {
+        transMeta = new TransMeta( fileAndPath );
+      } else {
+        transMeta = loadFromRepository( repository, fileAndPath );
+      }
+      String stepName = String.valueOf( parts[ 1 ].trim() );
+      DataServiceMeta dataServiceMeta = new DataServiceMeta( transMeta );
+      dataServiceMeta.setStepname( stepName );
+      dataServiceMeta.setName( dataServiceName );
+      PushDownOptimizationMeta pushDownMeta = new PushDownOptimizationMeta();
+      pushDownMeta.setStepName( stepName );
+      pushDownMeta.setType( cacheFactory.createPushDown() );
+      dataServiceMeta.setPushDownOptimizationMeta( Collections.singletonList( pushDownMeta ) );
+
+      return dataServiceMeta;
+    } catch ( Exception e ) {
+      return null;
+    }
+  }
+
+  private TransMeta loadFromRepository( Repository repository, String filePath ) throws KettleException {
+    String name = filePath.substring( filePath.lastIndexOf( "/" ) + 1, filePath.length() );
+    String path = filePath.substring( 0, filePath.lastIndexOf( "/" ) );
+
+    RepositoryDirectoryInterface root = repository.loadRepositoryDirectoryTree();
+    RepositoryDirectoryInterface rd = root.findDirectory( path );
+    if ( rd == null ) {
+      rd = root; // root
+    }
+    return repository.loadTransformation( repository.getTransformationID( name, rd ), null );
+  }
+
+  private boolean isTransient( String dataServiceName ) {
+    return dataServiceName.contains( DELIMITER );
+  }
```

Comment:
Yes, you are correct. This could potentially be problematic. Maybe I can add and additional identifier to the beginning that would be more unlikely for a user to do.